src/sage/topology/simplicial_complex.py

```diff
@@ -4802,7 +4810,110 @@ def intersection(self, other):
+            {(0, 0): 1, (-1, 4): 3, (-2, 6): 1, (-2, 8): 2, (-3, 10): 1}
+        """
+        if self._bbn is not None and 0 in self._bbn and base_ring in self._bbn[0]:
+            return self._bbn[base_ring]
+
+        from sage.homology.homology_group import HomologyGroup
+        L = self.vertices()
+        n = len(L)
+        B = {}
+        H0 = HomologyGroup(0, base_ring)
+
+        B[(0,0)] = ZZ.one()
+
+        for j in range(n+1):
+            for x in combinations(L, j):
+                S = self.generated_subcomplex(x)
+                H = S.homology(base_ring=base_ring)
+                for k in range(j):
+                    if j-k-1 in H and H[j-k-1] != H0:
+                        ind = (-k, 2*j)
+                        if ind not in B:
+                            B[ind] = ZZ.zero()
+                        B[ind] += len(H[j-k-1].gens())
+
+        if self._bbn is not None:
+            if 0 in self._bbn:
+                self._bbn[0].add(base_ring)
+            else:
+                self._bbn[0] = {base_ring}
+            self._bbn[base_ring] = B
+        else:
+            self._bbn = {0: {base_ring}, base_ring: B}
+        return B 
+
+    def bigraded_betti_number(self, a, b, base_ring=ZZ):
+        r"""
+        Return the bigraded Betti number indexed in the form `(-i, 2j)`.
+
+        Bigraded Betti number with indices `(-i, 2j)` is defined as a sum of ranks
+        of `(i-j-1)`-th (co)homologies of full subcomplexes with exactly `j` vertices. 
+
+        EXAMPLES::
+
+            sage: X = SimplicialComplex([[0,1],[1,2],[2,0],[1,2,3]])
+            sage: X.bigraded_betti_number(-1, 6)
+            1
+            sage: X.bigraded_betti_number(-1, 4, base_ring=QQ)
+            1
+            sage: X.bigraded_betti_numbers()
+            {(0, 0): 1, (-1, 4): 1, (-1, 6): 1, (-2, 8): 1}
+            sage: X.bigraded_betti_number(-2, 5)
+            0
+        """
+        if b % 2:
+            return base_ring.zero()
+        if a == 0 and b == 0:
+            return base_ring.one()
+        if self._bbn is not None and base_ring in self._bbn and (a,b) in self._bbn[base_ring]:
+            return self._bbn[base_ring].get((a,b), base_ring.zero())
+            
+        from sage.homology.homology_group import HomologyGroup
+
+        a = -a
+        b //= 2
+        L = self.vertices()
+        n = len(L)
+        H0 = HomologyGroup(0, base_ring)
+
+        B = 0
+
+        for x in combinations(L, b):
+            S = self.generated_subcomplex(x)
+            H = S.homology(base_ring=base_ring)
+            if b-a-1 in H and H[b-a-1] != H0:
+                B += len(H[b-a-1].gens())
+
+        res = ZZ(B)
+
+        if self._bbn is not None:
+            if base_ring in self._bbn:
+                self._bbn[base_ring][(-a, 2*b)] = res 
+            else:
+                self._bbn[base_ring] = {(-a, 2*b): res}
+        else:
+            self._bbn = {base_ring: {(-a, 2*b): res}}
 
+        return base_ring(B)
```

Comment:
```suggestion
        B = ZZ(B)

        if self._bbn is not None:
            if base_ring in self._bbn:
                self._bbn[base_ring][(-a, 2*b)] = B
            else:
                self._bbn[base_ring] = {(-a, 2*b): B}
        else:
            self._bbn = {base_ring: {(-a, 2*b): B}}

        return B
```
There's no need for the separate variable and this result should also not be in the `base_ring`.